src/base/bittorrent/session.cpp

```diff
@@ -4492,6 +4504,23 @@ void Session::handleFileErrorAlert(const lt::file_error_alert *p)
     m_recentErroredTorrentsTimer->start();
 }
 
+void Session::handleReadPieceAlert(const lt::read_piece_alert *p) const
+{
+#if (LIBTORRENT_VERSION_NUM < 10200)
+    if (p->ec) {
+        p->handle.auto_managed(false);
+        p->handle.pause();
```

Comment:
According to libtorrent docs:
>If the read failed, the torrent is paused and an error state is set

Seems you should just unset `seed_mode` flag.
Dunno why libtorrent doesn't do it itself...

Reply:
unsetting seed_mode flag doesn't do anything.
We have to unset auto_managed flag, otherwise after force_recheck the torrent will start downloading. This is because auto_managed auto starts torrents when errors get cleared.